src/tests/FlashOWare.Tool.Core.Tests/Analyzers/InterceptorGeneratorTests.cs

```diff
@@ -0,0 +1,122 @@
+using Basic.Reference.Assemblies;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Immutable;
+
+namespace FlashOWare.Tool.Core.Tests.Analyzers;
+
+public class InterceptorGeneratorTests
+{
+    private readonly Dictionary<string, string> _features;
+
+    private readonly ISourceGenerator _generator;
+    private readonly IEnumerable<ISourceGenerator> _generators;
+
+    public InterceptorGeneratorTests()
+    {
+        _features = new Dictionary<string, string>(1)
+        {
+            { "InterceptorsPreviewNamespaces", "AutoGenerated" },
+        };
+        _generator = new InterceptorGenerator().AsSourceGenerator();
+        _generators = ImmutableArray.Create(_generator);
+    }
+
+    [Fact]
+    public void Test()
```

Comment:
Soo...I dont love the naming here...but I can see why its tricky...potentially "CompilationWithInterceptors_GenerateInterceptableAttributeAndInterceptor_Success"
"CompilationWithInterceptors_RunGeneratorsAndUpdateCompilation_Success" 

I also don't love "success" but everything I come up with is even longer. 
P.S. Also I would love a session on all these changes. Though they generally look great to my (thank you so much for all your work) I am not sure I quite understand everything. 
P.P.S. Also I am not sure why I am writing in English ;-)

Reply:
good catch ... the name does not reveal anything ... I'm now thinking about
- Method under test: `RunGeneratorsAndUpdateCompilation` as it is in your 2nd suggestions ... strictly speaking, that's the method of the driver, that then invokes the `[Generator]`, but on the `[Generator]` it not only calls the single implemented method `Initialize(IncrementalGeneratorInitializationContext)` - strictly speaking - but also execute then the _"Incremental Steps"_ that the `[Generator]` is initializing ... so, IMO, your 2nd suggestion is the most revealing here
- Scenario: I suggest something like `InterceptableSourceText` ... I'll add a 2nd test that tests the negative `NonInterceptableSourceText` as well, where the `[Generator]` is expected to emit no source code
- Expected outcome: I like something in terms of your 1st suggestion: `GenerateInterceptableAttributeAndInterceptor`

re: P.S.
DM incoming ... perhaps also during our next live stream: https://www.youtube.com/watch?v=4XYxrCUj6k0

re: P.P.S.
for posterity, so that "the world" can read up on our discussion:
since this repository's visibility is _public_, everybody (including non-signed-in users) can see the Issues and Pull Requests of this repo, and also all discussions